app/scripts/modules/netflix/pipeline/stage/canary/canaryStage.js

```diff
@@ -33,6 +35,84 @@ module.exports = angular.module('spinnaker.netflix.pipeline.stage.canaryStage',
             stageTypes: ['bake', 'findAmi', 'findImage', 'findImageFromTags'],
             message: 'You must have a Bake or Find AMI stage before a canary stage.'
           },
+          {type: 'requiredField', fieldName: 'canary.canaryConfig.lifetimeHours', fieldLabel: 'Canary Lifetime'},
+          {type: 'requiredField', fieldName: 'baseline.account', fieldLabel: 'Account'},
+          {type: 'requiredField', fieldName: 'baseline.cluster', fieldLabel: 'Cluster'},
+          {type: 'requiredField', fieldName: 'clusterPairs', fieldLabel: 'Cluster Pairs'},
+          {type: 'requiredField', fieldName: 'canary.canaryConfig.canaryAnalysisConfig.name', fieldLabel: 'Configuration'},
+          {type: 'custom', fieldLabel: 'Lookback Duration', validate: (_pipeline, stage) => {
+            const cac = stage.canary.canaryConfig.canaryAnalysisConfig,
+              useLookback = cac.useLookback,
+              lookbackMins = cac.lookbackMins;
+            let result = null;
+            if (useLookback && (!lookbackMins || lookbackMins === 0)) {
+              result = 'When an analysis type of <strong>Sliding Lookback</strong> is selected, the lookback duration must be positive.';
+            }
+
+            return result;
+          }},
+          {type: 'custom', fieldLabel: 'Report Frequency', validate: (_pipeline, stage) => {
+            const reportFrequency = stage.canary.canaryConfig.canaryAnalysisConfig.canaryAnalysisIntervalMins;
+            let result = null;
+            if (!reportFrequency || (reportFrequency < 1)) {
+              result = 'The <strong>Report Frequency</strong> is required and must be positive.';
```

Comment:
Ha! Correct. Typo on my part. 0 < x < 1. Even with the min set to 1, I still like validators to do a complete validation, just in case someone changes the control, the validation of the underlying field still stands. The message implies that 0.5 is a valid value, so I'd want to validator to let that pass. Or change the message to '1 or greater' to properly capture the necessary value.